libraries/SITL/SIM_Aircraft.cpp

```diff
@@ -483,11 +483,19 @@ void Aircraft::fill_fdm(struct sitl_fdm &fdm)
 // is bouncing off:
 float Aircraft::perpendicular_distance_to_rangefinder_surface() const
 {
-    return sitl->height_agl;
+    switch ((Rotation)sitl->sonar_rot.get()) {
+    case Rotation::ROTATION_PITCH_270:
+        return sitl->height_agl;
+    case ROTATION_NONE ... ROTATION_YAW_315:
```

Comment:
how is this valid ?

Reply:
Magic!

Those are kept consecutive in our rotations list so we can get away with doing this....

The ellipsis is just "all enumeration entries between the two endpoints...